core/common/src/LocalDate.kt

```diff
@@ -19,6 +19,57 @@ import kotlinx.serialization.Serializable
  * The arithmetic on [LocalDate] values is defined independently of the time zone (so `2020-08-30` plus one day
  * is `2020-08-31` everywhere): see various [LocalDate.plus] and [LocalDate.minus] functions, as well
  * as [LocalDate.periodUntil] and various other [*until][LocalDate.daysUntil] functions.
+ *
+ * ### Arithmetic operations
+ *
+ * Operations with [DateTimeUnit.DateBased] and [DatePeriod] are provided for [LocalDate]:
+ * - [LocalDate.plus] and [LocalDate.minus] allow expressing concepts like "two months later,"
+ * - [LocalDate.until] and its shortcuts [LocalDate.daysUntil], [LocalDate.monthsUntil], and [LocalDate.yearsUntil]
+ *   can be used to find the number of days, months, or years between two dates,
+ * - [LocalDate.periodUntil] (and [LocalDate.minus] that accepts a [LocalDate])
+ *   can be used to find the [DatePeriod] between two dates.
+ *
+ * ### Construction, serialization, and deserialization
+ *
+ * [LocalDate] can be constructed directly from its components, using the constructor.
+ *
+ * ```
+ * LocalDate(year = 2023, monthNumber = 1, dayOfMonth = 2) == LocalDate(2023, Month.JANUARY, 2)
+ * ```
+ *
+ * [fromEpochDays] can be used to obtain a [LocalDate] from the number of days since the epoch day `1970-01-01`;
+ * [toEpochDays] is the inverse operation.
+ *
+ * ```
+ * LocalDate.fromEpochDays(0) == LocalDate(1970, Month.JANUARY, 1)
+ * LocalDate(1970, Month.JANUARY, 31).toEpochDays() == 30
+ * ```
+ *
+ * [parse] and [toString] methods can be used to obtain a [LocalDate] from and convert it to a string in the
+ * ISO 8601 extended format.
+ *
+ * ```
+ * LocalDate.parse("2023-01-02") == LocalDate(2023, Month.JANUARY, 2)
+ * LocalDate(2023, Month.JANUARY, 2).toString() == "2023-01-02"
+ * ```
+ *
+ * [parse] and [LocalDate.format] both support custom formats created with [Format] or defined in [Formats].
+ *
+ * ```
+ * val customFormat = LocalDate.Format {
```

Comment:
Comment to line `107` (`Format` builder) -- please keep it consistent with this one (i.e. `val myFormat = ...`)